packages/api/src/command/medical/document/document-bulk-download.ts

```diff
@@ -0,0 +1,170 @@
+import { DocumentReference } from "@medplum/fhirtypes";
+import { makeLambdaClient } from "../../../external/aws/lambda";
+import { getLambdaResultPayload } from "@metriport/core/external/aws/lambda";
+// import { getSignedUrls } from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+
+const BATCH_SIZE = 100;
+const lambdaClient = makeLambdaClient();
+const bulkSigningLambdaName = Config.getBulkUrlSigningLambdaName();
+
+export const triggerBulkUrlSigning = async (
+  cxId: string,
+  patientId: string
+): Promise<DocumentBulkDownloadProgress> => {
+  if (!bulkSigningLambdaName) throw new Error("Bulk Signing Lambda Name is undefined");
+  const { log } = Util.out(`triggerBulkSigning - M patient ${patientId}`);
+
+  const patient = await getPatientOrFail({ id: patientId, cxId });
+
+  const docBulkDownloadProgress = patient.data.documentBulkDownloadProgress;
+  const requestId = getOrGenerateRequestId(docBulkDownloadProgress);
+
+  if (isDocBulkDownloadProcessing(docBulkDownloadProgress?.download)) {
+    log(
+      `Patient ${patientId}, Request ${requestId}, docBulkDownloadProgress is already 'processing', skipping...`
+    );
+    return createQueryResponse("processing", patient);
+  }
+
+  const documents = await searchDocuments({ cxId, patientId });
+
+  let updatedPatient = await storeBulkDownloadQueryInit({
+    id: patient.id,
+    cxId: patient.cxId,
+    documentBulkDownloadProgress: { download: { status: "processing" } },
+    requestId,
+    totalDocuments: documents.length,
+  });
+
+  // sleep for 20 seconds to allow the webhook to be processed
+
+  let successes = 0;
+  const errors = 0;
+  // Process each batch
+  const batches = chunk(documents, BATCH_SIZE);
+  for (let batchIndex = 0; batchIndex < batches.length; batchIndex++) {
+    const batch = batches[batchIndex] as DocumentReference[];
+    const dtos: DocumentBulkDownloadDTO[] = [];
+
+    const payload = {
+      filenames: batch
+        .map(doc => {
+          if (doc.content && doc.content[0] && doc.content[0].attachment) {
+            return doc.content[0].attachment.title;
+          }
+          return undefined;
+        })
+        .filter((filename): filename is string => filename !== undefined),
+    };
+
+    // Invoke the lambda function
+    const result = await lambdaClient
+      .invoke({
+        FunctionName: bulkSigningLambdaName,
+        InvocationType: "RequestResponse",
+        Payload: JSON.stringify(payload),
+      })
+      .promise();
+
+    // TODO error handling logic and incrementing error count
+    const resultPayload = getLambdaResultPayload({ result, lambdaName: bulkSigningLambdaName });
+    const parsedResult: string[] = JSON.parse(resultPayload);
+
+    // local testing code
+    // const parsedResult: string[] = await getSignedUrls(
+    //   payload.filenames,
+    //   Config.getMedicalDocumentsBucketName(),
+    //   "us-east-2"
+    // );
+
+    // Create DTOs for each signed URL
```

Comment:
would be more efficient to generate dtos in lambda so only doing one loop through the filenames. we could move the dto definition to core and then use it in the lambda, but I didn't see any other dtos there, so thought it would be best to keep it with all the other dtos.